[Flang][OpenMP] OpenMP directive DEFAULT (NONE) is set, but a compilation-time error occurs indicating that the variable does not specify data-sharing attribute.

```
Version of flang-new : 20.0.0(2603c8b23be06116bc61bda36fae05b81d906c3f)/AArch64
```

When building [CP2K](https://www.cp2k.org/), the following error is detected.  
`OpenMP` directive `DEFAULT (NONE)` is set, but a compilation-time error occurs indicating that the variable does not specify `data-sharing` attribute.

The following are a self-made reproducer, Flang-new compilation result.

z1.f90:
```fortran
subroutine sub( aaa)
  real,dimension(*),intent(in)::aaa
  integer::ip
  real::ccc
!$omp parallel do private(ip,ccc) default(none)
  do ip = 1, 10
     ccc= aaa(ip)
  end do
end subroutine sub

program main
  real::arg(10)
  arg=1.0
  print *,'pass'
end program main
```

```
$ flang-new -fopenmp z1.f90
error: Semantic errors in z1.f90
./z1.f90:7:11: error: The DEFAULT(NONE) clause requires that 'aaa' must be listed in a data-sharing attribute clause
       ccc= aaa(ip)
            ^^^
$
```

--- a/flang/lib/Semantics/resolve-directives.cpp
+++ b/flang/lib/Semantics/resolve-directives.cpp
@@ -2052,6 +2052,8 @@ void OmpAttributeVisitor::Post(const parser::OpenMPAllocatorsConstruct &x) {
 static bool IsPrivatizable(const Symbol *sym) {
   auto *misc{sym->detailsIf<MiscDetails>()};
   return !IsProcedure(*sym) && !IsNamedConstant(*sym) &&
+      !semantics::IsAssumedSizeArray(
+          *sym) && /* OpenMP 5.2, 5.1.1: Assumed-size arrays are shared*/
       !sym->owner().IsDerivedType() &&
       sym->owner().kind() != Scope::Kind::ImpliedDos &&
       !sym->detailsIf<semantics::AssocEntityDetails>() &&

--- a/flang/test/Semantics/OpenMP/default-none.f90
+++ b/flang/test/Semantics/OpenMP/default-none.f90
@@ -47,3 +47,14 @@ subroutine sb4
     end do loop
   !$omp end parallel
 end subroutine
+
+! Test that default(none) does not error for assumed-size array
+subroutine sub( aaa)
+  real,dimension(*),intent(in)::aaa
+  integer::ip
+  real::ccc
+!$omp parallel do private(ip,ccc) default(none)
+  do ip = 1, 10
+     ccc= aaa(ip)
+  end do
+end subroutine sub

